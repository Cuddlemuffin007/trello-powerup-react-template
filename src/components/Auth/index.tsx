import React from "react";
import { Button, Message, Segment } from "semantic-ui-react";

import { useProvidedTrello } from "../../hooks";
import config from "../../config";

export const Auth = (): JSX.Element => {
    const t = useProvidedTrello();

    const handleAuthorize = async () => {
        const {
            trello: { tokenExpiration },
        } = config;
        await t.getRestApi().authorize({ scope: "read,write", expiration: tokenExpiration });
    };

    return (
        <Segment textAlign="center">
            <Message info content="Authorize your Trello account to continue." />
            <Button color="blue" content="Authorize" onClick={handleAuthorize} />
        </Segment>
    );
};
