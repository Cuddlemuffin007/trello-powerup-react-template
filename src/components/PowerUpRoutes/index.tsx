import React from "react";
import { Routes, Route } from "react-router-dom";
import { Message } from "semantic-ui-react";

import { Auth } from "../Auth";
import { CardButton } from "../../card-button/CardButton";

export const PowerUpRoutes = (): JSX.Element => {
    return (
        <Routes>
            <Route path="/" element={<Message compact info content="Nothing to see here!" />} />
            <Route path="/card-button.html" element={<CardButton />} />
            <Route path="/auth.html" element={<Auth />} />
        </Routes>
    );
};
