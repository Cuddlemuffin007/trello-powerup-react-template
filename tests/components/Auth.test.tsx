import React from "react";
import { shallow, mount } from "enzyme";
import { waitFor } from "@testing-library/react";

import { Auth } from "../../src/components/Auth";
import { TrelloProvider } from "../../src/components/TrelloProvider";
import { t } from "../helpers";

describe("Auth component", () => {
    beforeEach(() => {
        jest.resetModules();
    });

    it("should render the Auth component", () => {
        const app = shallow(<Auth />);
        expect(app).toMatchSnapshot();
    });

    it("should call the authorization handler using the trello iframe object provided", async () => {
        const mockAuthorize = jest.fn();
        (t.getRestApi as jest.Mock).mockImplementation(() => ({ authorize: mockAuthorize }));
        const app = mount(
            <TrelloProvider t={t}>
                <Auth />
            </TrelloProvider>,
        );
        app.find("button.ui.button.blue").simulate("click");
        await waitFor(() => {
            // "never" is default if not provided by process.env.TRELLO_TOKEN_EXPIRATION
            expect(mockAuthorize).toHaveBeenCalledWith({ scope: "read,write", expiration: "1hour" });
        });
        app.unmount();
    });
});
