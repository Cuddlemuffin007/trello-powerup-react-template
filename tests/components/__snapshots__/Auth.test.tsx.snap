// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth component should render the Auth component 1`] = `
<Segment
  textAlign="center"
>
  <Message
    content="Authorize your Trello account to continue."
    info={true}
  />
  <Button
    as="button"
    color="blue"
    content="Authorize"
    onClick={[Function]}
  />
</Segment>
`;
